qgreenland_dowload.py

```diff
@@ -349,6 +372,96 @@ def _fill_tree(self):
+            parent_item = None
+
+            while layer_hierarchy:
+                parent_text = layer_hierarchy[0]
+                del layer_hierarchy[0]
+
+                # try to find existing item for this group
+                if parent_item:
+                    start_index = self.list_model_manage.index(0, 0, self.list_model_manage.indexFromItem(parent_item))
+                else:
+                    start_index = self.list_model_manage.index(0,0, QModelIndex())
+                candidate_indices = self.list_model_manage.match(start_index, Qt.DisplayRole, parent_text, flags=Qt.MatchExactly)
+
+                if not candidate_indices:
+                    new_parent_item = QStandardItem(parent_text)
+                    new_parent_item.setFlags(Qt.ItemIsEnabled | Qt.ItemIsSelectable | Qt.ItemIsUserCheckable)
+                    new_parent_item.setCheckState(Qt.Unchecked)  # first columun, checkable, checked=0
+                    if parent_item:
+                        parent_item.appendRow([new_parent_item])
+                    else:
+                        self.list_model_manage.appendRow([new_parent_item])
+                    parent_item = new_parent_item
+                else:
+                    parent_item = self.list_model_manage.itemFromIndex(candidate_indices[0])
+
+            # create the child (checkable)
+            child = QStandardItem(layer['id'])
+            # set the custom data for the item as the unique id of each layer
+            child.setData(layer['id'], Qt.UserRole)
+            child.setFlags(Qt.ItemIsEnabled|Qt.ItemIsSelectable|Qt.ItemIsUserCheckable)
+            # to add an icon to the single item
+            # child.setIcon(0, QIcon(os.path.join(os.path.dirname(__file__), 'qgreenland-icon.png')))
+            child.setCheckState(Qt.Unchecked)
+
+            # add the icons depending on the checksum
+            # only if the json file in the profile folder exists
+            try:
+                for static_layer in self.downloaded_layers:
+                    for i in static_layer['assets']:
+                        # check if the layer of the manifest is in the static json file of the profile folder
+                        if layer['id'] in static_layer['id']:
+                            # get the data type - the layer
+                            if i['type'] == 'data':
+                                # if the checksum is the same - we have the most recent layer downloaded
+                                if layer['assets'][0]['checksum'] == i['checksum']:
+                                    child.setIcon(QIcon(os.path.join(os.path.dirname(__file__), 'icons','uptodate.png')))
+                                    child.setToolTip(self.tr("You already have the most recent data downloaded"))
+                                # if the checksum is not the same - warn the user with the specified icon
+                                elif layer['assets'][0]['checksum'] != i['checksum']:
```

Comment:
Nitpick: This line could just be `else:`

This is safer because when the code is maintained, someone might change line 449 and forget to make a corresponding change on this line.